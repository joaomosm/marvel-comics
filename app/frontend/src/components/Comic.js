import React from 'react';

const Comic = ({ comic, toggleFavorite, checkFavorite }) => {
  const isFavorite = checkFavorite(comic.id);
  const cssClasses = isFavorite ? 'image-overlay favorite' : 'image-overlay';

  return (
    <div key={comic.id} className='image-container' data-testid='comic'>
      <img src={`${comic.thumbnail.path}/portrait_xlarge.jpg`} alt='Avatar' className='image' data-testid='comic-thumbnail'/>
      <div className={cssClasses}>
        <div className='image-svg' onClick={() => toggleFavorite(comic)} />
        <div className='image-text' data-testid='comic-title'>{comic.title}</div>
      </div>
    </div>
  );
};

export default Comic;
