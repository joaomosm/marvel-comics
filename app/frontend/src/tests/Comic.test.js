import React from 'react';
import { render } from '@testing-library/react';

import Comic from './../components/Comic';

const defaultComic = {
  id: 1,
  title: 'le title',
  thumbnail: {
    path: ''
  }
}

const renderComponent = () =>
  render(<Comic comic={defaultComic} toggleFavorite={jest.fn()} checkFavorite={jest.fn()} />);

it('matches snapshot', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});

it('renders Comic component', () => {
  const { getByTestId } = renderComponent();
  expect(getByTestId('comic')).toBeVisible();
});

it('renders comic title', () => {
  const { getByTestId } = renderComponent();
  const element = getByTestId('comic-title')
  expect(element).toBeVisible();
  expect(element['innerHTML']).toBe(defaultComic.title)
});

it('renders comic thumbnail', () => {
  const { getByTestId } = renderComponent();
  const element = getByTestId('comic-thumbnail')
  expect(element).toBeVisible();
});

