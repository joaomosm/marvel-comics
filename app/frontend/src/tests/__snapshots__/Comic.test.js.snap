// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="image-container"
    data-testid="comic"
  >
    <img
      alt="Avatar"
      class="image"
      data-testid="comic-thumbnail"
      src="/portrait_xlarge.jpg"
    />
    <div
      class="image-overlay"
    >
      <div
        class="image-svg"
      />
      <div
        class="image-text"
        data-testid="comic-title"
      >
        le title
      </div>
    </div>
  </div>
</div>
`;
